### PR TITLE
Add group rest api user endpoints

### LINKED ISSUE #22387: Implement `group` endpoints
### Description

Group endpoints provides CRUD options for groups combined with assigning of additional resources i.e. users or roles.

* The endpoints should be implemented as described in [REST API Developer Guide](https://github.com/camunda/camunda/blob/main/docs/rest-controller.md)
* `{place-holder}` is the placeholder for parameters like `key`

### Endpoints

- [x] Create group - `POST /v2/groups` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.qc7j844hp0au)
- [x] Search groups - `POST /v2/groups/search` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.3r5lq4sb0tyu)
- [x] Get a single group - `GET /v2/groups/{key}` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.vsmunaey9342)
- [x] Update a group - `PATCH /v2/groups/{key}` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.27d3ji9wuci5)
- [x] Assign a user to a group - `POST /v2/groups/{key}/users/{userKey}` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.7d4q4ehhqx7d)
- [x] Search users in a group - `POST /v2/groups/{key}/users/search` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.1425quaqoc48)
- [x] Remove a user from a group - `DELETE /v2/groups/{key}/users/{userKey}` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.f2cf41s0545d)
- [x] Delete a group - `DELETE /v2/groups/{key}` - [API specification](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.oa81iyqm7b8)

### Acceptance Criteria

- [ ] Unit tests
- [ ] Integration tests

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Add Group REST API endpoints for user assign/unassign operations.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #22387

### CODE REVIEW

**Comment by koevskinikola** [comment]
@remcowesterhoud @nicpuppa sorry for chaining so many PRs, I can also merge them in a single, bigger PR, or base them off the first (create group) PR if you prefer that.

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
I think I made it through all your PRs 🎉

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -2664,6 +2664,50 @@ paths:
...
+                $ref: "#/components/schemas/ProblemDetail"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "404":
+          description: The group or user with the given key was not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "500":
```
[the remark this answers is not in the document]
Comment (koevskinikola) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2709]: 👍 I'll also add it to the REST API definition, as it's not documented there.

**Comment by koevskinikola** [comment]
@remcowesterhoud I applied your feedback.

**Review by remcowesterhoud — APPROVED** [review]
Thanks @koevskinikola 👍 

On a side note I just saw in the `GroupAddEntityProcessor` that we don't check for a duplicate assignment yet. I'll have to check the other processors as well, because if we don't do this check we'll run into consistency check error when we assign an entity multiple times

**Comment by koevskinikola** [comment]
@remcowesterhoud you added a duplicate check, but only on command distribution ([here](https://github.com/camunda/camunda/blob/e1008fc5d88e739c4ed459353fa09c5f0998157e/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/GroupAddEntityProcessor.java#L117-L125)). It's the same in the other Identity-related processors. 

Should I update them to do this check on the initial command processing?

**Comment by remcowesterhoud** [comment]
> @remcowesterhoud you added a duplicate check, but only on command distribution ([here](https://github.com/camunda/camunda/blob/e1008fc5d88e739c4ed459353fa09c5f0998157e/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/GroupAddEntityProcessor.java#L117-L125)). It's the same in the other Identity-related processors.
> 
> Should I update them to do this check on the initial command processing?

@koevskinikola That's not a duplicate check. Processing a distributed command must be idempotent. There's no guarantee that the second partition only receives this distributed command a single time. We need to be able to handle this by detecting if we already processed this command. That's what this check is for. If we detect we already added this entity we can reject the command, and acknowledge the distribution. This way the first partition is aware that the distribution was successful.

Without this check we'll run into an exception when we process the distirbuted command twice. That means there won't be an acknowledgement and the first partition will infinitely retry this distribution.

> Should I update them to do this check on the initial command processing?

The check should be added to the initial command processing additionally 🙂

**Comment by koevskinikola** [comment]
@remcowesterhoud I'll apply the check to roles, groups and tenants today.

**Pull request merged**